xskillscore/core/np_deterministic.py

```diff
@@ -116,17 +130,99 @@ def _pearson_r_p_value(a, b, weights, axis, skipna):
     """
     r = _pearson_r(a, b, weights, axis, skipna)
     a = np.rollaxis(a, axis)
-    df = a.shape[0] - 2
-    t_squared = r ** 2 * (df / ((1.0 - r) * (1.0 + r)))
-    _x = df / (df + t_squared)
+    b = np.rollaxis(b, axis)
+    # dirty fix: want to check the number of nans in union(a,b) on axis 0
```

Comment:
Is this still valid? It seems counts a union of the nan's is a workable solution

Reply:
I am not sure whether this is really want it should be doing. And I am confused because I dont see this in scipy.stats.pearson_r.

Reply:
I should take away the #dirty fix comment. I think this implementation is ok